### ISSUE TITLE
Failed to read area-describedby attribute

### ISSUE THREAD

**Issue by binayatdelhi** [nvaccess/nvda#9153]
NVDA is reading link content but failed to read area-describedby attribute if NVDA is configured with mouse tracking "on". It works perfectly with Tabbing.  

```
<a href="home.html" target="_blank" aria-describedby="refid" title="Home">Home</a>
<span hidden id="refid">Opens in new window</span>
```
NVDA version: 2018.4
Windows version: 7
Chrome: 71.0

**Comment by zstanecic** [comment]
@binayatdelhi,

Please fill the issue against the template

**Comment by Adriani90** [comment]
@binayatdelhi I think I can reproduce this issue in Chrome 74 or any other browser. The aria-describedby atribute is reported for me only when I press nvda+tab which reports the focused element. The aria-describedby is not reported when tabbing or arrowing in this case. This behavior does not depend on mouse tracking on or off. I get the same results.
@jcsteh is this caused by the span hidden ID maybe?

In this case  NVDA is expected to reportd the description "opens in a new window" when tabbing but it does not report it.

**Comment by carmacleod** [comment]
Still a problem in NVDA 2019.1.1
- I verified that "Report object descriptions" is selected in NVDA settings.
- I tried it with "mouse tracking" on and off, and it did not make any difference for me.

Tested in latest Chrome, latest Firefox. (did not test IE/Edge)

### Steps to reproduce:
Paste the following line into the browser's URL bar:
```
data:text/html,<a href="www.google.com" aria-describedby="opens-in-new-window">Google</a><span id="opens-in-new-window" hidden>Opens in a new window</span>
```

### Actual behavior:
When tabbing to this link, NVDA reads "link, Google" but does not read the description, "Opens in a new window".

### Expected behavior:
Expected to hear: "link, Google, Opens in a new window".

I agree that nvda+tab does read the description, but I expected the description to be read when tabbing as well (it is not).

### References:
This should be expected to work. Please note the following from [the AccName spec, step 2A](https://w3c.github.io/accname/#step2A):
> If the current node is hidden and is not directly referenced by aria-labelledby or aria-describedby, nor directly referenced by a native host language text alternative element (e.g. label in HTML) or attribute, return the empty string.
> Comment:
> By default, assistive technologies do not relay hidden information, but an author can explicitly override that and include hidden text as part of the accessible name or accessible description by using aria-labelledby or aria-describedby.

This [Short note on aria-labelledby and aria-describedby](https://developer.paciellogroup.com/blog/2015/05/short-note-on-aria-labelledby-and-aria-describedby/) by Steve Faulkner may add clarity.

**Comment by Adriani90** [comment]
cc: @MarcoZehe

**Comment by MarcoZehe** [comment]
NVDA has never reported the description, AccDescription, on occasions other than on focus or when asking for the focused element, or when navigating onto an object with object navigator. It never reports the description on links or most other elements with the virtual cursor. There are some occasions where NVDA 2019.3 will now do this, but the versions earlier than that won't. There is no bug here, and things work as expected if NVDA+Tab does report the description for you.

**Comment by Adriani90** [comment]
Closing as works for me, considering also Marco's comment above. Thanks.

**Comment by Adriani90** [comment]
Oh, after testing further, I will reopen this. The behavior is as follows:
Firefox: no Aria describedby reported when pressing nvda+tab or when navigating in focus mode, the link in the minimal test case is not reported at all and neither the aria describedby.

In Crome, the aria describedby is only reported when pressing nvda+tab, but not in focus mode.

In Edge 81 everything works correctly.

data:text/html,<a href="www.google.com" aria-describedby="opens-in-new-window">Google</a><span id="opens-in-new-window" hidden>Opens in a new window</span>

**Comment by Adriani90** [comment]
```
data:text/html,<a href="www.google.com" aria-describedby="opens-in-new-window">Google</a><span id="opens-in-new-window" hidden>Opens in a new window</span>
```

**Comment by Adriani90** [comment]
This might occur because it seems the span id is hidden. Is this expected behavior?

**Comment by rpearce** [comment]
I'm experiencing the same thing with NVDA with or without the hidden aspect in latest Edge (at least). For example, the `Read more` link on https://awkawk.github.io/aria-describedby_link.html (from https://www.w3.org/WAI/GL/wiki/Using_aria-describedby_for_link_purpose_-_May_2014), doesn't narrate the `aria-describedby` unless you tab to the link.

**Comment by leonardder** [comment]
@Adriani90 your example is incorrect. aria-describedby should contain the id of another html element that contains the description, not the description itself.

**Comment by rpearce** [comment]
> @Adriani90 your example is incorrect. aria-describedby should contain the id of another html element that contains the description, not the description itself.

His `aria-describedby` is `opens-in-new-window`, and the ID of the `<span>` is `id="opens-in-new-window"`, so I think that checks out.

**Comment by leonardder** [comment]
Ugh, I'm sorry. Really should have checked that with a braille display.

**Comment by Adriani90** [comment]
Same as #12718, closing this as duplicate because there is some moretechnical discussion in #12718.
This particular issue is not reproducible anymore when pressing tab, NVDA reports the description on tab press as well in focus mode. NVDA also reports it when tabbing in browse mode if the link is not the first or the last element of the virtual document.
However NVDA does not report descriptions when using quick navigation keys or arrow keys in browse mode. This can be discussed in the referenced issue.
